nixos/modules/services/audio/mpd.nix

```diff
@@ -136,67 +144,73 @@ in {
           parameter is omitted from the configuration.
         '';
       };
+
+      addBinariesToGlobalEnvironment = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to add the client and server binaries to the global environment.
+        '';
+      };
     };
 
   };
 
 
   ###### implementation
 
-  config = mkIf cfg.enable {
-
-    systemd.sockets.mpd = mkIf cfg.startWhenNeeded {
-      description = "Music Player Daemon Socket";
-      wantedBy = [ "sockets.target" ];
-      listenStreams = [
-        "${optionalString (cfg.network.listenAddress != "any") "${cfg.network.listenAddress}:"}${toString cfg.network.port}"
-      ];
-      socketConfig = {
-        Backlog = 5;
-        KeepAlive = true;
-        PassCredentials = true;
-      };
+  config = mkIf cfg.enable (let
+    description = "Music Player Daemon Socket";
+    listenStreams = [
+      "" # we need an empty string to reset the list instead of just adding
+      "%t/mpd/socket"
+      "${optionalString (cfg.network.listenAddress != "any") "${cfg.network.listenAddress}:"}${toString cfg.network.port}"
+    ];
+    wantedBy = [ "sockets.target" ];
+  in {
+    environment.systemPackages = lib.mkIf cfg.addBinariesToGlobalEnvironment (with pkgs; [ mpc_cli mpd ]);
+
+    systemd.packages = with pkgs; [ mpd ];
+
+    systemd.sockets.mpd = mkIf cfg.startSystemWide {
+      inherit description listenStreams wantedBy;
+    };
+
+    systemd.user.sockets.mpd = mkIf (!cfg.startSystemWide) {
```

Comment:
It might help to comment on what's going on with these sockets as it isn't immediately obvious how this works in conjunction with your addition to `systemd.packages`.